src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -2636,6 +2641,72 @@ void SurgeGUIEditor::showSettingsMenu(CRect &menuRect)
     eid++;
     mpeSubMenu->forget();
 
+    // Mouse behavior
+
+    int mid = 0;
+
+    COptionMenu* mouseSubMenu =
+        new COptionMenu(menuRect, 0, 0, 0, 0, VSTGUI::COptionMenu::kNoDrawStyle);
+
+    std::string mouseClassic = "Classic";
+    std::string mouseSlow = "Slow";
+    std::string mouseMedium = "Medium";
+    std::string mouseExact = "Exact";
+
+    addCallbackMenu(mouseSubMenu, mouseClassic.c_str(), [this]() {
+       CSurgeSlider::sliderMoveRateState = MoveRateState::kClassic;
+       Surge::Storage::updateUserDefaultValue(&(this->synth->storage), "sliderMoveRateState",
+                                              CSurgeSlider::sliderMoveRateState);
+    });
+    mid++;
+
+    addCallbackMenu(mouseSubMenu, mouseSlow.c_str(), [this]() {
+       CSurgeSlider::sliderMoveRateState = MoveRateState::kSlow;
+       Surge::Storage::updateUserDefaultValue(&(this->synth->storage), "sliderMoveRateState",
+                                              CSurgeSlider::sliderMoveRateState);
+    });
+    mid++;
+
+    addCallbackMenu(mouseSubMenu, mouseMedium.c_str(), [this]() {
+       CSurgeSlider::sliderMoveRateState = MoveRateState::kMedium;
+       Surge::Storage::updateUserDefaultValue(&(this->synth->storage), "sliderMoveRateState",
+                                              CSurgeSlider::sliderMoveRateState);
+    });
+    mid++;
+
+    addCallbackMenu(mouseSubMenu, mouseExact.c_str(), [this]() {
+       CSurgeSlider::sliderMoveRateState = MoveRateState::kExact;
+       Surge::Storage::updateUserDefaultValue(&(this->synth->storage), "sliderMoveRateState",
+                                              CSurgeSlider::sliderMoveRateState);
+    });
+    mid++;
+
+    std::string mouseMenuName = "Mouse Behavior ";
+
+    switch (CSurgeSlider::sliderMoveRateState)
+    {
+    case MoveRateState::kClassic:
+       mouseMenuName.append("(Classic)");
+       break;
+    case MoveRateState::kSlow:
+       mouseMenuName.append("(Slow)");
+       break;
+    case MoveRateState::kMedium:
+       mouseMenuName.append("(Medium)");
+       break;
+    case MoveRateState::kExact:
+       mouseMenuName.append("(Exact)");
+       break;
+    default:
+       break;
+    }
```

Comment:
This is a clever solution but the approach we've generally taken is to use menu checkmarks rather than menus this dynamic. I think the "Mouse Behavior (Slow)" menu would be a bit confusing compared to "Mouse Behavior >" and then "(check) Slow".  The only problem is addCallbackMenu doesn't let you set the check, but that's no problem. I'd suggest

1. Change the signature of addCallbackMenu to have a fourth argument `bool checked=false` and
2. Set that argument to the right boolean 

or

1. Change the signature of addCallbackMenu to return a CComandMenuItem *; and make it return menu then
2. Capture the return value in the parent

the VSTGUI API for checking a menu item is super simple. `menu->setChecked(true)`.

Reply:
Yes I agree again. I will improve it.

Reply:
Actually I like the 2nd choice better since it interferes less in menus without `kMultipleCheckStyle`.